src/test/kotlin/TestCreateNumalgo2.kt

```diff
@@ -0,0 +1,624 @@
+import org.dif.model.EncodingType
+import org.dif.model.PublicKeyAgreement
+import org.dif.model.PublicKeyAuthentication
+import org.dif.model.PublicKeyTypeAgreement
+import org.dif.model.PublicKeyTypeAuthentication
+import org.dif.peerdid.createPeerDIDNumalgo2
+import org.dif.peerdid.isPeerDID
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class TestCreateNumalgo2 {
+    @Test
+    fun testCreateNumalgo2Positive() {
+        val encryptionKeys = listOf(
+            PublicKeyAgreement(
+                encodedValue = "JhNWeSVLMYccCk7iopQW4guaSJTojqpMEELgSLhKwRr",
+                type = PublicKeyTypeAgreement.X25519, encodingType = EncodingType.BASE58
+            )
+        )
+        val signingKeys = listOf(
+            PublicKeyAuthentication(
+                encodedValue = "ByHnpUCFb1vAfh9CFZ8ZkmUZguURW8nSw889hy6rD8L7",
+                type = PublicKeyTypeAuthentication.ED25519, encodingType = EncodingType.BASE58
+            ),
+            PublicKeyAuthentication(
+                encodedValue = "3M5RCDjPTWPkKSN3sxUmmMqHbmRPegYP1tjcKyrDbt9J",
+                type = PublicKeyTypeAuthentication.ED25519, encodingType = EncodingType.BASE58
+            )
+        )
+        val service = """[
+        {
+            "type": "didcommmessaging",
+            "serviceEndpoint": "https://example.com/endpoint",
+            "routingKeys": ["did:example:somemediator#somekey"]
+        },
+        {
+            "type": "didcommmessaging",
+            "serviceEndpoint": "https://example.com/endpoint2",
+            "routingKeys": ["did:example:somemediator#somekey2"]
+        }
+        ]
+        """
+
+        val peerDIDAlgo2 = createPeerDIDNumalgo2(
+            encryptionKeys = encryptionKeys, signingKeys = signingKeys,
+            service = service
+        )
+        assert(
+            peerDIDAlgo2 == "did:peer:2.Ez6LSbysY2xFMRpGMhb7tFTLMpeuPRaqaWM1yECx2AtzE3KCc" +
+                ".Vz6MkqRYqQiSgvZQdnBytw86Qbs2ZWUkGv22od935YF4s8M7V" +
+                ".Vz6MkgoLTnTypo3tDRwCkZXSccTPHRLhF4ZnjhueYAFpEX6vg" +
+                ".SW3sidCI6ImRtIiwicyI6Imh0dHBzOi8vZXhhbXBsZS5jb20vZW5kcG9pbnQiLCJyIjpbImRpZDpleGFtcGxlO" +
+                "nNvbWVtZWRpYXRvciNzb21la2V5Il19LHsidCI6ImRtIiwicyI6Imh0dHBzOi8vZXhhbXBsZS5jb20vZW5kcG9p" +
+                "bnQyIiwiciI6WyJkaWQ6ZXhhbXBsZTpzb21lbWVkaWF0b3Ijc29tZWtleTIiXX1d"
+        )
+        assert(isPeerDID(peerDIDAlgo2))
+    }
+
+    @Test
+    fun testCreateNumalgo2WithoutEncryptionKeys() {
+        val encryptionKeys = listOf<PublicKeyAgreement>()
+        val signingKeys = listOf(
+            PublicKeyAuthentication(
```

Comment:
Although it's usually OK to have some copy-paste in tests to make it more visible, I think we can move repeating elements into a helper fixture (this can be done for unit tests only, not for the Demo scripts/tests where we need to set everything explicitly).
For example, we can have a list of valid keys, and a common service there.
After this our tests will look more compact (probably it's worth doing the same in Python when fixing service ID uniqueness issue):
```
@Test
fun test {
        val signingKeys = listOf(
            PublicKeyAuthentication(
                encodedValue = "....",
                type = PublicKeyTypeAuthentication.ED25519, encodingType = EncodingType.BASE58
            ),
            VALID_SIGNING_KEY1
        )
        assertThrows<IllegalArgumentException> {
            createPeerDIDNumalgo2(
                encryptionKeys =  listOf(VALID_X25519_KEY1, VALID_X25519_KEY2),
                signingKeys = signingKeys,
                service = VALID_SERVICE
            )
        }
}
```